OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationGMS.java

```diff
@@ -187,9 +192,13 @@ static void getLocation(Context context, boolean promptLocation, LocationHandler
          locationCoarse = true;
       }
 
+      String alertDialogTitle = context.getString(R.string.location_not_available_title);
+      String alertDialogMessage = context.getString(R.string.location_not_available_message);
+
       if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
          if (locationFinePermission != PackageManager.PERMISSION_GRANTED && locationCoarsePermission != PackageManager.PERMISSION_GRANTED) {
```

Comment:
It isn't clear here by the code but it can only fall under this if block if permission were missing from the `AndroidManifest.xml`

Reply:
This flow is added because the user might forget to add the permissions on Manifest, and location prompting will not work, because of that. So we are only showing this alert when the user is prompting from a preview.

Reply:
Sorry didn't finish my thought here. That isn't clear so we should add a comment.

The code originally should have had a comment about that for now it seems it needs it even more.

Reply:
Let me know if the changes make sense, in order to make it on iOS